<a href="https://colab.research.google.com/github/H2D2R2/Dz6/blob/main/%D0%94%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D0%B5%D0%B5_%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Задание 1

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [ ]:
import numpy as np


In [ ]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

sum(zp*ks) / len(zp)  -  (sum(zp)/len(zp)) * (sum(ks)/len(ks))                  #нахождение ковариации стандартным методом

9157.839999999997

In [ ]:
np.cov(zp,ks, ddof=0)                                                           #нахождение ковариации с помощью функции cov

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [ ]:
sum(zp)/len(zp)

101.4

In [ ]:
sum(ks)/len(ks)

709.9

In [ ]:
9157.84 / (101.4*709.9)                                                         #нахождение коэффициента Пирсона с помощью библиотеки numpy

0.1272207408716343

In [ ]:
(sum(zp*ks) / len(zp)  -  (sum(zp)/len(zp)) * (sum(ks)/len(ks))) / ((sum(zp)/len(zp)) * (sum(ks)/len(ks)))     #нахождение коэффициента Пирсона простым способом 

0.12722074087163424

Ответ: Ковариация: 9157.84. Коэффициент корреляции Пирсона: 0.1272.

# Задание 2

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [ ]:
a = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
t = 2.262
n = 10
sigma = np.std(a, ddof=1)
x = np.mean(a)

A0 = x - t*(sigma/np.sqrt(n))
A0

110.55660776308164

In [ ]:
A1 = x + t*(sigma/np.sqrt(n))
A1

125.64339223691834

Ответ: {110.5;125.6}

# Задание 3

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [ ]:
M = 174.2
z = 1.96
D = 25
sigma = np.sqrt(D)                                                                   
n = 27

A0 = M - (z * sigma/np.sqrt(n))                                                
A0

172.31398912064722

In [ ]:
A1 = M + (z * sigma/np.sqrt(n)) 
A1

176.08601087935276

Ответ: {172;176}

# Задание 4 

Тему для курсовой выберу чуть позже и согласую с Вами в личных сообщениях. 